main.cpp

```diff
@@ -19,20 +19,33 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        const auto *cast_expression = Result.Nodes.getNodeAs<CStyleCastexpr>("cast");
+        auto replace = CharSourceRange::getCharRange(cast_expression->getLParenLoc(),
+                            cast_expression->getSubexprAsWritten()->getBeginLoc());

```

Comment:
Please describe the difference between `getSubExprAsWritten()->getBeginLoc()` and `getRParenLoc().getLocWithOffset(1)`.

Reply:
getSubExprAsWritten()->getBeginLoc() points to the beginning of the name of the variable that we're trying to cast
getRParenLoc().getLocWithOffset(1) points to the character that's right after the closing parenthesis of the type name to which we're trying to cast to
Which means if we use it, the spaces after the closing parenthesis will transfer with the name of the variable

For example:

int main() {
    float foo;
    int i = (int)                     foo ;
    return 0;
}
Using getSubExprAsWritten()->getBeginLoc():

int main() {
    float foo;
    int i = static_cast<int>(foo) ;
    return 0;
}

And if we use getRParenLoc().getLocWithOffset(1):

int main() {
    float foo;
    int i = static_cast<int>(                     foo) ;
    return 0;
}

Using getSubExprAsWritten()->getBeginLoc() is more appropriate.